snomed/com.b2international.snowowl.snomed.datastore/src/com/b2international/snowowl/snomed/datastore/request/SnomedRefSetMemberCreateRequest.java

```diff
@@ -170,6 +172,18 @@ public Set<String> getRequiredComponentIds(TransactionContext context) {
 
 	@Override
 	public String execute(TransactionContext context) {
+		// generate id field as UUID if not defined otherwise
+		if (Strings.isNullOrEmpty(id)) {
+			id = UUID.randomUUID().toString();
+		} else {
+			// otherwise validate that it is a valid UUID
+			try {
+				UUID.fromString(id);
+			} catch (IllegalArgumentException e) {
+				throw new BadRequestException("Refset Member uses an illegal non-UUID identifier '%s'.", id);
```

Comment:
If the message is user-facing, we might need the entire incantation:

```suggestion
				throw new BadRequestException("Reference set member uses an illegal non-UUID identifier '%s'.", id);
```